frontend/packages/core/src/card.tsx

```diff
@@ -126,20 +119,90 @@ const BaseCardContent = styled.div<SpacingProps>`
   ${spacing}
 `;
 
-const StyledCardContent = styled(BaseCardContent)({
+const StyledCardContentContainer = styled.div({
   "> .MuiPaper-root": {
     border: "0",
     borderRadius: "0",
   },
+  overflow: "hidden",
 });
 
+const BaseCardActionArea = styled(CardActionArea)<SpacingProps>`
+  ${spacing}
+`;
+
+const StyledCardActionArea = styled(BaseCardActionArea)({
+  ":hover": {
+    backgroundColor: "#F5F6FD",
+  },
+
+  ":active": {
+    backgroundColor: "#D7DAF6",
+  },
+});
+
+const StyledExpandButton = styled(IconButton)({
+  width: "32px",
+  height: "32px",
+  color: "#3548D4",
+});
+
+interface CardContentCollapsibleProps {
+  title: string;
+  dualStateIcon?: { true: React.ReactElement; false: React.ReactElement };
+}
+
 interface CardContentProps extends SpacingProps {
   children?: React.ReactNode | React.ReactNode[];
+  collapsible?: boolean;
+  collapseAction?: CardContentCollapsibleProps;
+  maxHeight?: number;
 }
 
-const CardContent = ({ children, ...props }: CardContentProps) => (
-  <StyledCardContent {...props}>{children}</StyledCardContent>
-);
+const CardContent = ({
+  children,
+  collapsible = false,
+  collapseAction,
+  maxHeight,
+  ...props
+}: CardContentProps) => {
+  const ref = React.useRef(null);
+  const [showExpand, setShowExpand] = React.useState(false);
+  const [expanded, setExpanded] = React.useState(true);
+
+  const defaultMaxHeight = maxHeight || "none";
+
+  React.useEffect(() => {
+    if (ref.current.scrollHeight > defaultMaxHeight) {
+      setShowExpand(true);
+      setExpanded(false);
+    }
+  }, [defaultMaxHeight]);
+
+  return (
+    <BaseCardContent {...props} ref={ref}>
+      <StyledCardContentContainer style={{ maxHeight: expanded ? "none" : defaultMaxHeight }}>
+        {children}
+      </StyledCardContentContainer>
+      {collapsible && showExpand && (
+        <StyledCardActionArea padding={1} onClick={() => setExpanded(!expanded)}>
```

Comment:
This isn't on the figma design but I think the entire "see more" area should be a button instead of require the users to have to explicitly click on the IconButton. Lmk what you think